context_prop/src/main/java/io/opentelemetry/context/propagation/HttpTextFormat.java

```diff
@@ -88,17 +107,18 @@
   /**
    * Extracts the value from upstream. For example, as http headers.
    *
-   * <p>If the value could not be parsed, the underlying implementation will decide to return an
-   * object representing either an empty value, an invalid value, or a valid value. Implementation
-   * must not return {@code null}.
+   * <p>If the value could not be parsed, the underlying implementation will decide to set an object
+   * representing either an empty value, an invalid value, or a valid value. Implementation must not
+   * set {@code null}.
    *
+   * @param context the {@code Context} used to store the extracted value.
```

Comment:
The javadoc is not clear whether context should be modified in-place or not. If it should not, I suggest

```suggestion
   * @param parentContext the {@code Context} from which to create a child in which the extracted value will be stored.
```

or something like that.

Reply:
I suggest we do as in the Specification, and clarify at the top level docs that `Context` is immutable (grpc's Context is explicit about this, we can add a note here too).

Reply:
As of now, `Context` stands as an immutable object in the Specification.